### ISSUE TITLE
peer._lastServerId is not being set & peer.reconnect() deletes peer.id

### ISSUE THREAD

**Issue by RefinedSoftwareLLC** [peers/peerjs#265]
//peer._lastServerId is not being set when I use: 
peer = new Peer(null, {host: '165.225.157.206', port: 9000})
// or
peer = new Peer({host: '165.225.157.206', port: 9000})
console.log(peer._lastServerId)

//This makes it so that when 

peer.reconnect()

//is used, it connects you with the id of "undefined" or "null"

//Current Fix:

var last_id = null
var peer = new Peer(null, {host: '165.225.157.206', port: 9000})
peer.on('open', peer_open) // (id)
peer.on('disconnected', peer_disconnected) // ()

function peer_open (id) {
    if (id === null) {
        peer.id = last_id
    } else {
        last_id = id
    }
}

function peer_disconnected () {
    peer.id = last_id
    peer._lastServerId = last_id
    peer.reconnect()
}

**Comment by fyyyyy** [comment]
i have this problem

```js
 VIDEO peer.open My peer ID is: KU_kjH8ZYbjfAPJ2AAAA
peer.js:1117 PeerJS:  ERROR Error: Lost connection to server.
Video.js:55 Lost connection to server.Video.js:55 (anonymous function)peer.js:708 EventEmitter.emitpeer.js:1736 Peer.emitErrorpeer.js:1481 (anonymous function)peer.js:705 EventEmitter.emitpeer.js:2595 _http.onreadystatechangepeer.js:2568 (anonymous function)
peer.js:1117 PeerJS:  ERROR Aborting!
peer.js:1117 PeerJS:  ERROR Error: ID `KU_kjH8ZYbjfAPJ2AAAA` is taken
peer.js:1117 PeerJS:  ERROR Error: Lost connection to server.

**Comment by afrokick** [comment]
I'll try to reproduce it and write a test and fix it if it still exists

**Comment by alexbonair** [comment]
Looks like this one still exists with 1.1.0. 
In the Demo (https://github.com/jmcker/Peer-to-Peer-Cue-System) linked on peerjs.com, I found this workaround:
`// Workaround for peer.reconnect deleting previous id
if (peer.id === null) {
    console.log('Received null id from peer open');
    peer.id = lastPeerId;
} else {
    lastPeerId = peer.id;
}`
But as peer.id is readonly, that's not valid option. 
Any other workarounds or fixes for this reconnect() problem?

**Comment by afrokick** [comment]
You can help us) create PR if you have a fix please.

**Comment by afrokick** [comment]
fixed in #580
